docs/introduction/3_frame_forwarding.md

```diff
@@ -18,3 +16,13 @@ Protocol frames often have specific rules governing their behaviour, including i
 Traffic frames, on the other hand, are always forwarded using SNEK routing or tree routing (typically the former) and are not required to be otherwise inspected by an intermediate node.
 
 If a suitable next-hop is identified, the frame will be forwarded to the chosen next-hop peer.
+
+## Watermarks
+
+In the case of **traffic frames** there is also a watermark used to help detect routing loops. Watermarks only factor into the equation when next-hops are selected from the SNEK routing table as these routes can become stale much more quickly than route information obtained through the global spanning tree. The watermarks are designed to track the closest destination node information that any given node along the route that a packet traverses knows about. If any node along the route does not know of either the same best destination node or a closer destination node, then the packet has reached a location where further forwarding could result in routing loops.
```

Comment:
> The watermarks are designed to track the closest destination node information that any given node along the route that a packet traverses knows about.

It might be more concise to say:

> The watermark ensures that forward progress towards a given key is being made and that a packet will never be forwarded onto a path that is worse than the path selected at the last hop.

I also find it makes more rational sense in my mind to think about "paths" rather than "nodes", since ultimately we're comparing the watermark to path information.